Add AI/BI dashboard example (#43)
databricks/bundle-examples: knowledge_base/dashboard_nyc_taxi/README.md

# Dashboard for NYC Taxi Trip Analysis

This example demonstrates how to model an AI/BI dashboard in a Databricks Asset Bundle.

It includes and deploys the sample __NYC Taxi Trip Analysis__ dashboard to a Databricks workspace.

> For more information about AI/BI dashboards, please refer to the [documentation](https://docs.databricks.com/en/dashboards/index.html#dashboards).

Reviewer: ```suggestion
For more information about AI/BI dashboards, please refer to the [documentation](https://docs.databricks.com/en/dashboards/index.html).
```
Anchor is not needed as this is the main dashboards article.

## Prerequisites

* Databricks CLI v0.232.0 or above

> ## Usage

Reviewer: Should the steps in here be numbered?

Author: We could but then should apply it in the other examples in the knowledge base as well.

What do you typically convey with numbering over plain enumeration?

> First:

Reviewer: ```suggestion
Modify `databricks.yml`:
```

> * Update the `host` field under `workspace` in `databricks.yml` to the Databricks workspace you wish to deploy to.

Reviewer: ```suggestion
* Update the `host` field under `workspace` to the Databricks workspace to deploy to.
```

> * Update the `warehouse` field under the `warehouse_id` variable to match the SQL warehouse you wish to use.

Reviewer: ```suggestion
* Update the `warehouse` field under `warehouse_id` to the name of the SQL warehouse to use.
```

Run `databricks bundle deploy` to deploy the dashboard.

> Run `databricks bundle open` to open the deployed dashboard in your browser.
> 
> Alternatively, run `databricks bundle summary` to display its URL.

Reviewer: ```suggestion
Run `databricks bundle open` to navigate to the deployed dashboard in your browser. Alternatively, run `databricks bundle summary` to display its URL.
```

### Visual modification

> Dashboards are inherently visual, so you may want to use the web interface to modify the dashboard.
> Any modifications that are done through the web interface won't automatically be reflected in the `.lvdash.json` file.
> You can, however, run a dedicated bundle command to retrieve these changes and update your `.lvdash.json` file.

Reviewer: ```suggestion
You can use the Databricks UI to modify the dashboard, but any modifications made through the UI will not be applied to the bundle `.lvdash.json` file unless you run `bundle generate` or `bundle deploy --force`.
```

> To retrieve the updated `.lvdash.json` file, run:

Reviewer: ```suggestion
To update the local bundle `.lvdash.json` file, run:
```

Reviewer: (Hmm....maybe original is better...)

```sh
databricks bundle generate dashboard --resource nyc_taxi_trip_analysis --force
```

To continuously poll and retrieve the updated `.lvdash.json` file when it changes, run:

```sh
databricks bundle generate dashboard --resource nyc_taxi_trip_analysis --force --watch
```

Any remote modifications of a dashboard are noticed by the `deploy` command and require

> you to explicitly acknowledge that remote changes are overwritten by local changes.

Reviewer: ```suggestion
you to explicitly acknowledge that remote changes can be overwritten by local changes.
```
